From cbeede06e08420087ef2db5873922452b6e78a51 Mon Sep 17 00:00:00 2001
From: Mark Charlebois <charlebm@gmail.com>
Date: Thu, 6 Mar 2014 11:26:07 -0800
Subject: [PATCH] module: LLVMLinux: Remove unused function warning from

Converts a compile time type check that generates an unused function
warning to an unused variable warning which is suppressed when
building the kernel.

This code cannot be upstreamed as it is because it causes GCC to report
an equal number of unused-variable warnings.

Author: Mark Charlebois <charlebm@gmail.com>
Signed-off-by: Mark Charlebois <charlebm@gmail.com>
Signed-off-by: Behan Webster <behanw@converseincode.com>
---
 include/linux/moduleparam.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/moduleparam.h b/include/linux/moduleparam.h
index c3eb102..f0cf1a2 100644
--- a/include/linux/moduleparam.h
+++ b/include/linux/moduleparam.h
@@ -346,7 +346,8 @@ static inline void destroy_params(const struct kernel_param *params,
 /* The macros to do compile-time type checking stolen from Jakub
    Jelinek, who IIRC came up with this idea for the 2.4 module init code. */
 #define __param_check(name, p, type) \
-	static inline type *__check_##name(void) { return(p); }
+	static inline type *__check_##name(void) { return(p); } \
+	static void *silence_warning_##name = (void *)__check_##name;
 
 extern struct kernel_param_ops param_ops_byte;
 extern int param_set_byte(const char *val, const struct kernel_param *kp);
-- 
1.8.3.2

